service/r2/service.go

```diff
@@ -96,6 +99,39 @@ func newServiceMetrics(scope tally.Scope, samplingRate float64) serviceMetrics {
 	}
 }
 
+type route struct {
```

Comment:
super nit: might wanna consider moving `route` to the bottom so the service related pieces are in one piece.